Improved release process (#95)
heroku/languages-github-actions: README.md

- [Workflows](#workflows)
  - [Buildpacks - Prepare Release](#buildpacks---prepare-release)
  - [Buildpacks - Prepare Release](#buildpacks---release)
- [Actions](#actions)
  - [Install Languages CLI](#install-languages-cli)
- [Development](#development)

## Workflows

### Buildpacks - Prepare Release

Prepares a buildpack release by:
- bumping the fixed version
- updating changelogs
- generating an aggregate changelog from all the changelogs
- opening a PR against the repository with the modified files

You can pin to:

> - the [latest release](/releases/latest) version with `latest`
> - a [specific release](/releases) version with `@v{major}.{minor}.{patch}`
> - the development version with `main`

Reviewer: If I click the links when having opened the `README.md` directly (as opposed to viewing when looking at the GitHub overview UI), they currently 404. 

eg Visit https://github.com/heroku/languages-github-actions/blob/06ac54789d81ade15ace09a6e1fa18ae1860b659/README.md and click the links.

I think the only way to avoid that is having to include the full URL sadly.

Also, currently some of the examples are prefixed with `@` but some aren't - I think for clarity/consistency we should prefix them all with `@`.

```suggestion
- the [latest release](https://github.com/heroku/languages-github-actions/releases/latest) version with `@latest`
- a [specific release](https://github.com/heroku/languages-github-actions/releases) version with `@v{major}.{minor}.{patch}`
- the development version with `@main`
```

jobs:
  prepare-release:
    uses: heroku/languages-github-actions/.github/workflows/_buildpacks-prepare-release.yml@latest
    with:

>       app_id: ${{ vars.GH_APP_ID }}

Reviewer: I presume this is `vars.GH_APP_ID` instead of `vars.LINGUIST_GH_APP_ID` (and similar for the secrets section) since you're wanting to keep the README generic, and not tied to our specific Linguist implementation choices? (I'm happy with either approach, so how it is at the moment is fine - I'm only mentioning in case this naming wasn't intentional)

Author: yeah, it was to keep the README generic

#### Inputs

| Name                            | Description                                                                                                                                                                                             | Required | Default                     |
|---------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------|-----------------------------|
| `app_id`                        | Application ID of GitHub application (e.g. the Linguist App)                                                                                                                                            | true     |                             |

> | `bump`                          | Bump which coordinate? (major, minor, patch)                                                                                                                                                            | true     |                             |

Reviewer: ```suggestion
| `bump`                          | Which component of the version to increment (major, minor, or patch)                                                                                                                                                            | true     |                             |
```

### Buildpacks - Release

> Prepares one or more buildpacks release by:

Reviewer: I think this sentence needs adjusting, since it mentions preparing a release, but this workflow is the "perform a release" one?

Author: done

* Detects all the buildpacks in a repository and compiles them into Cloud Native Buildpacks
* For each compiled buildpack:
  * Creates a CNB archive file from the compiled buildpack and publishes it as a GitHub Release

>   * Creates an OCI image from the compiled buildpack and publishes it to Docker

Reviewer: (Feel free to reword this suggestion further - it's still missing an explanation of `[metadata.release]` etc, though maybe that's good enough for now?)

```suggestion
  * Creates an OCI image from the compiled buildpack and publishes it to the Docker Hub repository specified in the buildpack's `buildpack.toml`
```

Author: added information about the metadata key used

>   * Retrieves the OCI image published to Docker and registers this with the CNB Registry

Reviewer: ```suggestion
  * Retrieves the OCI image published to Docker Hub and registers this with the CNB Registry
```

> * Once all buildpacks have been published, all the buildpack references found in [heroku/builder](https://github.com/heroku/builder)

Reviewer: ```suggestion
* Once all buildpacks have been published, updates all the buildpack references found in [heroku/builder](https://github.com/heroku/builder)
```

  for the given list of builders and opens a pull requests containing all the changes to be committed.

You can pin to:

> - the [latest release](/releases/latest) version with `latest`
> - a [specific release](/releases) version with `@v{major}.{minor}.{patch}`
> - the development version with `main`

Reviewer: ```suggestion
- the [latest release](https://github.com/heroku/languages-github-actions/releases/latest) version with `@latest`
- a [specific release](https://github.com/heroku/languages-github-actions/releases) version with `@v{major}.{minor}.{patch}`
- the development version with `@main`
```

## Actions

### Install Languages CLI

> Downloads the Languages CLI from a known release or installs using Cargo

Reviewer: ```suggestion
Downloads the Languages CLI from a known release or installs using Cargo.
```

#### Usage

```yaml
- name: Install Languages CLI
  uses: heroku/languages-github-actions/.github/actions/install-languages-cli@latest
```

You can pin to:

> - the [latest release](/releases/latest) version with `latest`
> - a [specific release](/releases) version with `@v{major}.{minor}.{patch}`
> - the development version with `main`

Reviewer: ```suggestion
- the [latest release](https://github.com/heroku/languages-github-actions/releases/latest) version with `@latest`
- a [specific release](https://github.com/heroku/languages-github-actions/releases) version with `@v{major}.{minor}.{patch}`
- the development version with `@main`
```
